system/Validation/FormatRules.php

```diff
@@ -44,7 +44,7 @@ public function alpha_space(?string $value = null): bool
 			return true;
 		}
 
-		return (bool) preg_match('/^[A-Z ]+$/i', $value);
+		return (bool) preg_match('/\A[A-Z ]+\z/i', $value);
```

Comment:
please add `@see https://regex101.com/` link comment so it will be easier to verify in the future if bug/improvement needed

Reply:
@samsonasik 

> please add @see https://regex101.com/ link comment so it will be easier to verify in the future if bug/improvement needed

Do we write this link in the code where all the regular expressions are written?
I don't see the need to add a link.
I think the attached URL is maintained by an individual, but how long will it be valid?

Reply:
You can check these:

https://github.com/codeigniter4/CodeIgniter4/blob/3119fc8d0b8790cc408c292bcab218931f6d32d2/system/View/Parser.php#L576

https://github.com/codeigniter4/CodeIgniter4/blob/3119fc8d0b8790cc408c292bcab218931f6d32d2/system/View/Parser.php#L742

https://github.com/codeigniter4/CodeIgniter4/blob/3119fc8d0b8790cc408c292bcab218931f6d32d2/utils/Rector/UnderscoreToCamelCaseVariableNameRector.php#L28

Yes, not everwhere yet, but I think it will be better to add it whenever possible for better future.

I've no information about how long it will be valid, but it seems it long enough ,for example, this is from 5 years ago https://regex101.com/library/wL5mP7 .

Reply:
Will this link really be easy to maintain?
Every time a regular expression is edited, does the URL also need to be maintained, and will all the maintainers review and manage it?

The system also has the ability to delete the URLs that are created.
After the merge, if the link is deleted, the information will be lost.
I think test code is more constructive than a link that may or may not work.

What do other MEMBERS think about this?
I don't want to add this URL as my commit, but I will reluctantly add it if many members agree with this.

Reply:
I vote for adding the link.
As it is positioned directly above the code I would think it is implicit to maintain it (or at least see what it does) when changing the regex itself.
Also, while of course the link might point to a dead end some future day, I consider it to be better to provide information that might get lost instead of not providing info at all in the first place.

Reply:
It would be nice to have a link. It will make our life easier in the future.

Reply:
Thanks for giving me your input.
I have added your comment.